mk/config.mk

```diff
@@ -909,15 +909,28 @@ CFG_REGULATOR_FIXED ?= n
 $(eval $(call cfg-enable-all-depends,CFG_REGULATOR_FIXED, \
 	 CFG_DRIVERS_REGULATOR CFG_DT))
 
-# The purpose of this flag is to show a print when booting up the device that
-# indicates whether the board runs a standard developer configuration or not.
+# When enabled, CFG_INSECURE permits insecure configuration of OP-TEE core
+# and shows a print (info level) when booting up the device that
+# indicates that the board runs a standard developer configuration.
+#
 # A developer configuration doesn't necessarily has to be secure. The intention
 # is that the one making products based on OP-TEE should override this flag in
 # plat-xxx/conf.mk for the platform they're basing their products on after
 # they've finalized implementing stubbed functionality (see OP-TEE
 # documentation/Porting guidelines) as well as vendor specific security
 # configuration.
-CFG_WARN_INSECURE ?= y
+CFG_INSECURE ?= y
+
+# CFG_WARN_INSECURE serves the same purpose as CFG_INSECURE but is deprecated
+# and should not be used. CFG_WARN_INSECURE will be removed in a future OP-TEE
+# release.
+ifneq (,$(CFG_WARN_INSECURE))
+$(info CFG_WARN_INSECURE is deprecated, please use CFG_INSECURE instead)
+else
+ifneq ($(CFG_WARN_INSECURE),$(CFG_INSECURE))
```

Comment:
This doesn't seem very useful. If someone is using `CFG_WARN_INSECURE` in a downstream tree set to `n` we'll error out here. If the downstream tree doesn't touch `CFG_WARN_INSECURE` or otherwise leaves it at its default value we'll pass.

If the objective is to catch deprecated usage of `CFG_WARN_INSECURE`, then we should use "error" instead of "info" above.

Reply:
Ok, I'll simplify this to the below (I hope I understood your proposal):
```patch
 ifneq (,$(CFG_WARN_INSECURE))
-$(info CFG_WARN_INSECURE is deprecated, please use CFG_INSECURE instead)
+$(error CFG_WARN_INSECURE is deprecated, please use CFG_INSECURE instead)
-ifneq ($(CFG_WARN_INSECURE),$(CFG_INSECURE))
-$(error Deprecated CFG_WARN_INSECURE must match CFG_INSECURE)
-endif
 endif
```

Reply:
close this discussion thread, superseded by discussion thread related to https://github.com/OP-TEE/optee_os/pull/6457#discussion_r1392595513.